scripts/deprecate-npm-package.ts

```diff
@@ -0,0 +1,96 @@
+import execa from "execa";
+
+interface DeprecateOptions {
+  deprecateMessage: string;
+  packageNamesToBeDeprecated: string[];
+  versionsToBeDeprecated: RegExp;
+  isProduction?: boolean;
+}
+
+/**
+ * Options for deprecating npm packages by using this script
+ * `npm deprecate` (https://docs.npmjs.com/cli/v8/commands/npm-deprecate) supports using semver for specifying versions.
+ *  However, canary version cannot be deprecated easily.
+ *  e.g., `~1.0.0-0` will match:
+ *  - `1.0.0-xyz` (canary version)
+ *  - `1.0.x` (stable version)
+ *  but NOT 1.0.1-xyz (canary version)
+ *  Therefore, to match both versions below and you need a union version (~1.0.0-0 || ~1.0.1-0) or multiple commands:
+ *  - `1.0.0-xyz`
+ *  - `1.0.1-xyz`
+ *  We should not use multiple commands, as the server has certain limit.
+ */
+const deprecateOptions: DeprecateOptions = {
+  /**
+   * Deprecate message for all packages/versions to be deprecated.
+   */
+  deprecateMessage: "1.x is no longer maintained.",
```

Comment:
Perhaps use a bit longer message like - `1.x is no longer maintained. Please migrate to the latest version of the SAP Cloud SDK for JS.`

Reply:
The packages were deprecated.
This PR is for documentation purpose and can be reused in the future.
So this message is just an example.